manifests/instance_service.pp

```diff
@@ -0,0 +1,89 @@
+# == Define: haproxy::instance_service
+#
+# Set up the environment for an haproxy service.
+#   * Associate an haproxy instance with the haproxy package it should use.
+#   * Create the start/restart/stop functions needed by Service[].
+# In other words: sets things up so that Service[$instance_name] will work.
+#
+# In particular:
+#
+# * Create a link to the binary an instance will be using. This
+#   way each instance can link to a different binary.
+#   If you have an instance called "foo", you know "haproxy-foo"
+#   is a link to the binary it should be using.
+# * Create an init.d file named after the instance. This way
+#   Service[$instance] can start/restart the service.
```

Comment:
Short answer: That's how we do it at Stack.

Longer answer:   As the next paragraph says, this module is just one example of how to do this. Other sites can choose to do it other ways.  All you have to do is set up Service[$instance_name], and there are many other ways is can be done.